### ISSUE TITLE
"Submit as JCL" in file explorer displayed for every file

### ISSUE THREAD

**Issue by phaumer** [zowe/vscode-extension-for-zowe#2574]
**Describe the bug**

The context menu in the VS Code File Explorer is already very busy. Many other extension contribute to it. It is essential for a good user experience that it only shows operations that are valid in the current context of the selected file. 

Recently a new entry was added by Zowe Explorer to show "Submit as JCL" for every file in the explorer. As it also included group separator at the top and bottom the vertical real estate it uses is significant. Zowe Explorer was intended to be language agnostic and purely a tool that focuses on the interactions with z/OS independent of the content of the text files it displays. It therefore should not assume any language ids as those would be contributed by language extensions. So ideally a JCL editing extension that integrated with Zowe Explorer should contribute such a menu and bind it to a jcl language id "when" clause in the menu definition.

I propose to either remove the menu completely and let developers of JCL editor implement it or let the user configure a file extension (or list a third party language id) that should be used for the menu's when clause. If no extension is configured then the menu should not be created as third party extension might already provide it.

**To Reproduce**

1. Go to file explorer
2. Right-click any file that is not a JCL file

<img width="389" alt="2023-11-14_13-48-07" src="https://github.com/zowe/vscode-extension-for-zowe/assets/4391934/f9043b6c-4bb3-4e08-9f87-815a3a60f90f">

**Expected behavior**

Zowe Explorer should be language independent.

**Desktop (please complete the following information):**

- OS: any 
- Zowe Explorer Version: 2.12.1
- (Optional) Zowe CLI Version:
- (Optional) Are you using Secure Credential Store?

**Additional context**

**Comment by oseam87** [comment]
I agree, I always see this menu item (and I never used it). No, I don't want to submit my program as JCL ;-)

**Comment by zFernand0** [comment]
one way we could resolve this is by adding JCL as a language provided by Zowe Explorer.
That way we can add the check back that only adds this option to JCL files.
We just need to investigate the possible drawbacks of doing this.

```jsonc
{
  // ...
  "contributes": {
    "languages": [
      { "id": "jsonc",  "filenamePatterns": [ "zowe.config*.json" ] },
      { "id": "jcl",    "filenamePatterns": [ "*.jcl" ] }
    ]
  }
}
```

**Comment by phaumer** [comment]
@zFernand0 my thinking was that there would be a user setting for Zowe Explorer that allows specifying a language id to use for the where clause, because different JCL language extension could use different language ids as we seen this for COBOL out there. One extension might be using "jcl" versus another could be using "JCL" or "JobControlLanguage". In other words Zowe Explorer should not make any assumptions, but allow to work with any of these by configuration.

**Comment by t1m0thyj** [comment]
In addition to restricting the "Submit as JCL" command to show only for JCL files, we may want to move it further down the list. Right now it appears near the top when right-clicking any file in VS Code file explorer or right-clicking an editor:
![image](https://github.com/user-attachments/assets/bef26b87-0f74-4807-8a76-b6ee45908414)
